src/cryptography/hazmat/primitives/asymmetric/ec.py

```diff
@@ -109,99 +197,253 @@ def public_bytes(self, encoding, format):
 
 
 @utils.register_interface(EllipticCurve)
-class SECT571R1(object):
+class SECT571R1(EllipticCurve):
     name = "sect571r1"
     key_size = 571
+    field = EllipticCurveBinaryField(571, 2, 5, 10)
+    a = 1
+    b = int("02f40e7e2221f295de297117b7f3d62f5c6a97ffcb8ceff1"
+            "cd6ba8ce4a9a18ad84ffabbd8efa59332be7ad6756a66e29"
+            "4afd185a78ff12aa520e4de739baca0c7ffeff7f2955727a", 16)
+    x = int("0303001d34b856296c16c0d40d3cd7750a93d1d2955fa80a"
+            "a5f40fc8db7b2abdbde53950f4c0d293cdd711a35b67fb14"
+            "99ae60038614f1394abfa3b4c850d927e1e7769c8eec2d19", 16)
+    y = int("037bf27342da639b6dccfffeb73d69d78c6c27a6009cbbca"
+            "1980f8533921e8a684423e43bab08a576291af8f461bb2a8"
+            "b3531d2f0485c19b16e2f1516e23dd3c1a4827af1b8ac15b", 16)
+    n = int("03ffffffffffffffffffffffffffffffffffffffffffffff"
+            "ffffffffffffffffffffffffe661ce18ff55987308059b18"
+            "6823851ec7dd9ca1161de93d5174d66e8382e9bb2fe84e47", 16)
 
 
 @utils.register_interface(EllipticCurve)
-class SECT409R1(object):
+class SECT409R1(EllipticCurve):
     name = "sect409r1"
     key_size = 409
+    field = EllipticCurveBinaryField(409, 87)
```

Comment:
I'm currently taking advantage of *args here to allow a variable number of polynomial exponents without requiring ugly extra parentheses and commas. A keyword version of this would end up looking like:

    field = EllipticCurveBinaryField(m=409, f=(87,))

I can do that, but I didn't think it really improved the readability.